core/config/README.md

```diff
@@ -0,0 +1,91 @@
+# Configuration Package
+
+At a high level, configuration is any data that is used in an application but
+not part of the application itself. Any reasonable complex system will need to
+have knobs to tune and not everything can have intelligent defaults.
+
+In UberFx, we try very hard to make sure that users only need to provide the
+minimal amount of configuration necessary to have working components, while also
+allowing them to override any field they need to if the defaults don't make
+sense for them.
+
+## Provider
+
+`ConfigurationProvider` is the interface for anything that can provide values.
+We provide a few reference implementations (environment and YAML), but you are
+free to register your own providers via `config.RegisterProviders()` and
+`config.RegisterDynamicProviders`.
+
+## Dynamic Configuration Providers
+
+Dynamic configuration providers are treated specially because they frequently
+need some bootstrap configuration to be useful. They conform to the same
+`ConfigurationProvider` interface but are instantiated **after** the static
+providers, so that they can read static values. For example, if you were to
+implement a ZooKeeper-backed `ConfigurationProvider`, you'd likely need to
+specify (via YAML or environment variables) where your ZooKeeper nodes live.
+
+## ConfigurationValue
+
+`ConfigurationValue` is the return type of all configuration providers'
+`GetValue(key string)` method. Under the hood, we use the empty interface
+(`interface{}`) since we don't necessarily know the structure of your
+configuration ahead of time.
+
+There are two main usages for a `ConfigurationValue`:
```

Comment:
Can rephrase for clarity about who's using what, like:

You can use a `ConfigurationValue` for two main purposes:

or 

A `ConfigurationValue` can serve two main purposes: